### PR TITLE
[Feature] Introduce meta spec interface

### LINKED ISSUE #43460: Iceberg metadata super optimization
- [x] https://github.com/StarRocks/starrocks/pull/45640
## Enhancement
This issue is used to trace iceberg metadata related optimization patches. The first version patch is https://github.com/StarRocks/starrocks/pull/43459 

There are currently four important optimizations:
1. iceberg metadata distributed plan.   （Performance increased by n ~ 2n times. n is be number）
2. iceberg manifest cache.  (Job planning delay reduced to around 100ms)
3. external table based on metadata file distributed plan Framework.
4. refactor some iceberg metadata parts.

There is still a lot of optimization work to come, some of which are as follows:

- [ ]  Add more auto distributed strategies baed on be load like cpu/io/mem
- [ ]  Enhancement time travel statement like `as of`.
- [ ] refactor current split structure like `RemoteFileInfo`
- [ ] provide default optimal distributed plan threshold by do a lot of testing. 
- [ ] Adapt to partition data parser when exists partition evolution.
- [ ] Continuously optimize performance
- [ ] add cache metrics like hitrate
- [ ] provide benchmark to guarantee performance

### PR DESCRIPTION
**Pull request by stephen-shelby**
## What I'm doing:

Fixes #issue
https://github.com/StarRocks/starrocks/issues/43460

## What type of PR is this:

- [ ] BugFix
- [x] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.3
  - [ ] 3.2
  - [ ] 3.1
  - [ ] 3.0
  - [ ] 2.5

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 1]
```diff
@@ -439,6 +450,70 @@ public boolean prepareMetadata(MetaPreparationItem item, Tracers tracers) {
...
+        if (fileIO instanceof IcebergCachingFileIO) {
+            fileIO = ((IcebergCachingFileIO) fileIO).getWrappedIO();
+        }
+
+        String serializedTable = SerializationUtil.serializeToBase64(new SerializableTable(nativeTable, fileIO));
+
+        return new IcebergMetaSpec(serializedTable, remoteMetaSplits);
+    }
+
+    private List<ManifestFile> filterManifests(List<ManifestFile> manifests,
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:492]: There are repetitive methods ？

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java [hunk 2]
```diff
@@ -439,6 +450,70 @@ public boolean prepareMetadata(MetaPreparationItem item, Tracers tracers) {
...
+                                               org.apache.iceberg.Table table, Expression filter) {
+        Map<Integer, ManifestEvaluator> evalCache = specCache(table, filter);
+
+        return manifests.stream()
+                .filter(manifest -> manifest.hasAddedFiles() || manifest.hasExistingFiles())
+                .filter(manifest -> evalCache.get(manifest.partitionSpecId()).eval(manifest))
+                .collect(Collectors.toList());
+    }
+
+    private Map<Integer, ManifestEvaluator> specCache(org.apache.iceberg.Table table, Expression filter) {
```
Comment (Youngwb) [inline fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java:502]: It also look like has repetitive methods

Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java]: resolved

File: fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java (hunk 1 above)
Comment (stephen-shelby) [reply fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java]: resolved

**Review by Youngwb — APPROVED** [review]

**Review by dirtysalt — APPROVED** [review]

**Pull request merged**